code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -143,3 +144,48 @@ export function composeStories<TModule extends Store_CSFExports>(
 
   return composedStories;
 }
+
+declare global {
+  type WrappedStoryRef = { __pw_type: 'jsx' | 'importRef' };
+  type UnwrappedJSXStoryRef = {
+    __pw_type: 'jsx';
+    type: ComposedStoryFn;
+  };
+  type UnwrappedImportStoryRef = ComposedStoryFn;
+
+  function __pwUnwrapObject(
+    storyRef: WrappedStoryRef
+  ): UnwrappedJSXStoryRef | UnwrappedImportStoryRef;
+}
+
+export function createPlaywrightTest<TFixture extends { extend: any }>(
+  baseTest: TFixture
+): TFixture {
+  return baseTest.extend({
+    mount: async ({ mount, page }: any, use: any) => {
+      await use(async (storyRef: WrappedStoryRef) => {
+        // load the story in the browser
+        await page.evaluate(async (wrappedStoryRef: WrappedStoryRef) => {
+          const unwrappedStoryRef = await globalThis.__pwUnwrapObject?.(wrappedStoryRef);
+          const story =
+            '__pw_type' in unwrappedStoryRef ? unwrappedStoryRef.type : unwrappedStoryRef;
+          return story?.load?.();
+        }, storyRef);
+
+        // mount the story
+        const mountResult = await mount(storyRef);
+
+        // play the story in the browser
+        await page.evaluate(async (wrappedStoryRef: WrappedStoryRef) => {
+          const unwrappedStoryRef = await globalThis.__pwUnwrapObject?.(wrappedStoryRef);
+          const story =
+            '__pw_type' in unwrappedStoryRef ? unwrappedStoryRef.type : unwrappedStoryRef;
+          const canvasElement = document.querySelector('#root');
+          return story?.play?.({ canvasElement });
```

Comment:
We're being very defensive here with optional chaining on `__pwUnwrapObject` and `story`. The upside is that we're using experimental APIs so they could change and break at any point in time, which would not cause this to crash. The downside is that it would silently error for anyone.

WDYT @yannbf?

Reply:
Maybe we add a warning/log message mentioning that users should open a GH issue?